src/factorizations/orthnull.jl

```diff
@@ -0,0 +1,96 @@
+using MatrixAlgebraKit:
+  MatrixAlgebraKit,
+  left_orth_polar!,
+  left_orth_qr!,
+  left_orth_svd!,
+  left_polar!,
+  lq_compact!,
+  qr_compact!,
+  right_orth_lq!,
+  right_orth_polar!,
+  right_orth_svd!,
+  right_polar!,
+  select_algorithm,
+  svd_compact!
+
+function MatrixAlgebraKit.left_orth!(
```

Comment:
Is this not caught by the MatrixAlgebraKit implementation? If not, it probably should?

Reply:
It's a bit subtle, the MatrixAlgebraKit.jl version is designed around the output has being pre-allocated, so it has an extra argument for the output. That is tricky to support for block sparse matrices since the different backends (SVD, QR/LQ, polar) might have different block structures/sectors so it isn't straightforward to pre-allocate the output in that way.

We can move this code over to MatrixAlgebraKit.jl, that was the main thing I wanted your feedback on.

Reply:
Ok, in that case I'll make a PR to MatrixAlgebraKit.jl, it would be nice to not have this logic here.